components/motor/gpio/motor_encoder.go

```diff
@@ -597,6 +549,7 @@ func (m *EncodedMotor) Close(ctx context.Context) error {
 		return err
 	}
 	m.cancel()
+	m.rpmMonitorDone()
```

Comment:
cc @martha-johnston this fixed the tests by cancelling the goroutine, but maybe it is better to call this from stop.